inst/rmd/html_report.Rmd

```diff
@@ -18,120 +18,45 @@ knitr::opts_chunk$set(
   warning = FALSE
 )
 
-library(ggplot2)
-
 # We expect a rooms object as input.
-
 rooms <- params$rooms
 
 if (is.null(rooms)) {
   stop("No data provided!")
 }
-
-data <- rooms$events
-
-end <- max(data$time)
-
-data <- data |>
-  dplyr::filter(time > end - lubridate::days(30)) |>
-  dplyr::filter(!is.na(body) | type == "m.reaction") |> # chat only
-  dplyr::mutate(
-    day = lubridate::date(time),
-    hour = lubridate::hour(time)
-  )
 ```
 
 Statistics generated on `r Sys.time()`
 
-This data has a total of **`r length(unique(data$sender))`** different IDs,
-speaking a total of **`r nrow(data)`** messages.
+This data has a total of **`r length(unique(rooms$events$sender))`** different
+IDs, speaking a total of **`r nrow(rooms$events)`** messages.
 
-### Daily Activity & trend (last 30 days)
```

Comment:
Any particular reason why you're dropping the 30 day filter here?

We pass `rooms` rather than `rooms$events` to the report, so it's not trivial to filter the event list before calling the report. Perhaps we should allow a filter YAML parameter instead?

Reply:
I thought that the data would either be prefiltered as desired or only reach back to the time. But going back to it, I think the functions for plotting should accept the time filter parameter with a default value.

Reply:
Yeah, that makes sense. I was seeing some weirdness in my `get_rooms` calls after merging #16 that I am not seeing with these patches, so I think I'm going to merge this in the next hour or so (to keep HEAD working cleanly). We can roll-forward on adding time filters to the plots.

Reply:
I updated the branch reintroducing the time limit. But as it applies to all plots, I added it to the introducing paragraphs. I think that this makes more sense, but we can also go back to the initial visuals.

Reply:
Oh, thanks! I wasn't trying to make you rush, I would have been happy to merge as is and then fix this later - but this is great. I agree we probably need to revisit the report itself - the new member calls mean we should probably add some of that data too.

Reply:
No problem!